nni/algorithms/compression/v2/pytorch/utils/evaluator.py

```diff
@@ -536,88 +538,90 @@ def get_dummy_input(self) -> Any:
-    It has some requirements for the writing of the training loop, please refer to the documentation for details.
+    TorchEvaluator is the Evaluator for native PyTorch users.
+    Please refer to the :ref:`compression-evaluator` for the evaluator initialization example.
 
     Parameters
     ----------
     training_func
         The training function is used to train the model, note that this a entire optimization training loop.
-        It should have three required parameters [model, optimizers, criterion]
-        and three optional parameters [schedulers, max_steps, max_epochs].
-        ``optimizers`` can be an instance of ``torch.optim.Optimizer`` or a list of ``torch.optim.Optimizer``,
-        it belongs to the ``optimizers`` pass to ``TorchEvaluator``.
-        ``criterion`` and ``schedulers`` are also belonging to the ``criterion`` and ``schedulers`` pass to ``TorchEvaluator``.
-        ``max_steps`` and ``max_epochs`` are used to control the training duration.
-
-        Example::
-
-            def training_func(model: Module, optimizer: Optimizer, criterion: Callable, scheduler: _LRScheduler,
-                              max_steps: int | None = None, max_epochs: int | None = None, *args, **kwargs):
-                model.train()
-
-                # prepare data
-                data_dir = Path(__file__).parent / 'data'
-                MNIST(data_dir, train=True, download=True)
-                transform = transforms.Compose([transforms.ToTensor(), transforms.Normalize((0.1307,), (0.3081,))])
-                mnist_train = MNIST(data_dir, train=True, transform=transform)
-                train_dataloader = DataLoader(mnist_train, batch_size=32)
-
-                max_epochs = max_epochs if max_epochs else 3
-                max_steps = max_steps if max_steps else 6000
+        Training function has three required parameters, ``model``, ``optimizers`` and ``criterion``,
+        and three optional parameters, ``lr_schedulers``, ``max_steps``, ``max_epochs``.
+
+        Let's explain these six parameters NNI passed in, but in most cases, users don't need to care about these.
+        Users only need to treat these six parameters as the original parameters during the training process.
+
+        * The ``model`` is a wrapped model from the original model, it has a similar structure to the model to be pruned,
+          so it can share training function with the original model.
+        * ``optimizers`` are re-initialized according to ``optimizers`` passed to the evaluator and the wrapped model's parameters.
+        * ``criterion`` also based on the ``criterion`` passed to the evaluator, it might be modified in some algorithms.
+        * If users use ``lr_schedulers`` in the ``training_func``, NNI will re-initialize the ``lr_schedulers`` with the re-initialized
+          optimizers.
+        * ``max_steps`` is the NNI training duration limitation. An integer means that after ``max_steps`` steps, the training should stop.
```

Comment:
-> is for pruner (or quantizer) to control the number of training steps. The user implemented ``training_func`` should respect ``max_steps`` by stopping the training loop after ``max_steps`` is reached.